accession-commons-monotonic-generator-jpa/src/test/java/uk/ac/ebi/ampt2d/test/configuration/TestBasicRestControllerForDecoratedAccession.java

```diff
@@ -0,0 +1,123 @@
+import org.springframework.boot.test.autoconfigure.orm.jpa.AutoConfigureDataJpa;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+import uk.ac.ebi.ampt2d.commons.accession.autoconfigure.EnableBasicRestControllerAdvice;
+import uk.ac.ebi.ampt2d.commons.accession.autoconfigure.EnableDecoratedAccessionMergedExceptionAspect;
+import uk.ac.ebi.ampt2d.commons.accession.autoconfigure.EnableSpringDataContiguousIdService;
+import uk.ac.ebi.ampt2d.commons.accession.core.AccessioningService;
+import uk.ac.ebi.ampt2d.commons.accession.core.BasicAccessioningService;
+import uk.ac.ebi.ampt2d.commons.accession.core.DecoratedAccessioningService;
+import uk.ac.ebi.ampt2d.commons.accession.generators.monotonic.MonotonicAccessionGenerator;
+import uk.ac.ebi.ampt2d.commons.accession.hashing.SHA1HashingFunction;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.monotonic.service.ContiguousIdBlockService;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.repositories.BasicJpaAccessionedObjectCustomRepositoryImpl;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.service.BasicJpaInactiveAccessionService;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.repositories.IAccessionedObjectCustomRepository;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.services.InactiveAccessionService;
+import uk.ac.ebi.ampt2d.commons.accession.service.BasicSpringDataRepositoryMonotonicDatabaseService;
+import uk.ac.ebi.ampt2d.test.models.TestModel;
+import uk.ac.ebi.ampt2d.test.persistence.TestLongHistoryRepository;
+import uk.ac.ebi.ampt2d.test.persistence.TestLongOperationEntity;
+import uk.ac.ebi.ampt2d.test.persistence.TestMonotonicEntity;
+import uk.ac.ebi.ampt2d.test.persistence.TestMonotonicInactiveAccessionEntity;
+import uk.ac.ebi.ampt2d.test.persistence.TestMonotonicInactiveAccessionRepository;
+import uk.ac.ebi.ampt2d.test.persistence.TestMonotonicRepository;
+import uk.ac.ebi.ampt2d.test.rest.TestControllerForDecoratedAccession;
+
+@Configuration
+@EnableSpringDataContiguousIdService
+@EnableWebMvc
+@EnableBasicRestControllerAdvice
+@EnableDecoratedAccessionMergedExceptionAspect
+@ComponentScan(basePackageClasses = IAccessionedObjectCustomRepository.class)
+@EnableJpaAuditing
+@ComponentScan(basePackageClasses = BasicJpaAccessionedObjectCustomRepositoryImpl.class)
+@EntityScan("uk.ac.ebi.ampt2d.test.persistence")
+@EnableJpaRepositories(basePackages = {"uk.ac.ebi.ampt2d.test.persistence",
+        "uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.monotonic.repositories"
+})
+@AutoConfigureDataJpa
+public class TestBasicRestControllerForDecoratedAccession {
```

Comment:
`BasicRestControllerForDecoratedAccessionTestConfiguration` seems more appropriate to me.